src/main/kotlin/at/bitfire/dav4jvm/property/webdav/GetETag.kt

```diff
@@ -46,19 +46,24 @@ class GetETag(
            weak       = "W/"
            opaque-tag = quoted-string
         */
-        val tag: String
-
-        // remove trailing "W/"
-        if (rawETag.startsWith("W/") && rawETag.length >= 2) {
-            // entity tag is weak
-            tag = rawETag.substring(2)
-            weak = true
+        val tag: String?
+
+        if (rawETag != null) {
+            // remove trailing "W/"
+            if (rawETag.startsWith("W/") && rawETag.length >= 2) {
+                // entity tag is weak
+                tag = rawETag.substring(2)
+                weak = true
+            } else {
+                tag = rawETag
+                weak = false
+            }
         } else {
-            tag = rawETag
+            tag = null
             weak = false
         }
 
-        eTag = QuotedStringUtils.decodeQuotedString(tag)
```

Comment:
It's not really hard to understand, but I have to think about all possibilities (when is it null, when not?) when deciding whether this is correct.

So my style suggestion because it's easier and less prone to errors:

```
if (rawETag != null) {
    val tag: String

    // remove trailing "W/"
    if (rawETag.startsWith("W/")) {
        // entity tag is weak
        tag = rawETag.substring(2)
        weak = true
    } else {
        tag = rawETag
        weak = false
    }

    eTag = QuotedStringUtils.decodeQuotedString(tag)
} else {
    eTag = null
    weak = false
}
```

Then it's clear when decode is called and when not and there are no `?` and `::` constructs.

Hope that it's still correct, please re-check of course ;)